pageserver/src/tenant/timeline/offload.rs

```diff
@@ -19,6 +19,9 @@ pub(crate) async fn offload_timeline(
         return Ok(());
     };
 
+    // Now that the Timeline is in Stopping state, request all the related tasks to shut down.
+    timeline.shutdown(super::ShutdownMode::Hard).await;
```

Comment:
Yeah, it is a significant bug fix. The test would fail without it; it would be timing out, waiting for the pageserver to shut down.

Interestingly enough, the pageserver shuts down without problems if you "just" do an offload. If you then do an unoffload however, the pageserver hangs during shutdown.

The reason why I didn't put it into a separate PR was that it would be a single line change only, and ideally comes with a regression test, which this PR already serves for. Do you think it's okay if I mention it in the frontmatter of the PR?

Reply:
Probably fine, your call. I'm not sure to what extent we communicate changes via commit logs -- this would be more important if we were doing e.g. backports or it was a risky change that someone would have to search through the commit log to see/bisect.

Reply:
Edited the commit msg